projects/ad2s1210_iio/app/ad2s1210_iio.c

```diff
@@ -0,0 +1,627 @@
+ * @param       readval[out]- Pointer to variable to read data into
+ * @return      0 in case of success, negative value otherwise
+ */
+static int32_t iio_ad2s1210_debug_reg_read(void *dev, uint32_t reg,
+		uint32_t *val)
+{
+	return ad2s1210_reg_read(dev,  (uint8_t) reg, (uint8_t *)val);
+}
+
+/*!
+ * @brief       Write the debug register value
+ * @param       dev[in, out]- Pointer to IIO device instance
+ * @param       reg[in]- Register address to read from
+ * @param       writeval[out]- Pointer to variable to write data into
+ * @return      0 in case of success, negative value otherwise
+ */
+static int32_t iio_ad2s1210_debug_reg_write(void *dev, uint32_t reg,
+		uint32_t val)
+{
+	return ad2s1210_reg_write(dev, (uint8_t)reg, (uint8_t)val);
+}
+
+/**
+ * @brief	Read buffer data corresponding to AD2S1210 IIO device
+ * @param	iio_dev_data[in] - Pointer to IIO device data structure
+ * @return	0 in case of success, negative error code otherwise
+ */
+#if (DATA_CAPTURE_MODE == BURST_DATA_CAPTURE)
+static int32_t iio_ad2s1210_submit_buffer(struct iio_device_data *iio_dev_data)
+{
+	uint32_t sample_indx = 0;
+	uint32_t nb_of_samples;
+	int32_t ret;
+	uint16_t data[2];
+	uint8_t i;
+
+	nb_of_samples = iio_dev_data->buffer->size / BYTES_PER_SAMPLE;
+
+	if (!buf_size_updated) {
+		iio_dev_data->buffer->buf->size = iio_dev_data->buffer->size;
+		buf_size_updated = true;
+	}
+
+	/* Push into circular buffer */
+	while (sample_indx < nb_of_samples) {
+		ret = ad2s1210_spi_single_conversion(ad2s1210_dev_inst,
+						     iio_dev_data->buffer->active_mask,
+						     data, sizeof(data));
+		if (ret) {
+			return ret;
+		}
+		ret = no_os_cb_write(iio_dev_data->buffer->buf, data,
+				     BYTES_PER_SAMPLE * active_count);
+		if (ret) {
+			return ret;
+		}
+		sample_indx++;
+	}
+	return 0;
+}
+#else
+static int32_t iio_ad2s1210_submit_buffer(struct iio_device_data *iio_dev_data)
+{
+	return 0;
+}
+#endif
+
+/**
+ * @brief	Prepare for RESOLVER data capture (from device to memory)
+ * @param	dev_instance[in] - IIO device instance
+ * @param	chn_mask[in] - Channels select mask
+ * @return	0 in case of success, negative error code otherwise
+ */
+static int32_t iio_ad2s1210_prepare_transfer(void *dev_instance,
+		uint32_t chn_mask)
+{
+#if (DATA_CAPTURE_MODE == CONTINUOUS_DATA_CAPTURE)
+	int32_t ret;
+
+	ret = iio_trig_enable(ad2s1210_hw_trig_desc);
+	if (ret) {
+		return ret;
+	}
+#endif
+	active_count = __builtin_popcount(chn_mask);
+
+	return 0;
+}
+
+/**
+ * @brief	Perform tasks before end of current data transfer
+ * @param	dev[in] - IIO device instance
+ * @return	0 in case of success, negative error code otherwise
+ */
+static int32_t iio_ad2s1210_end_transfer(void *dev)
+{
+#if (DATA_CAPTURE_MODE == CONTINUOUS_DATA_CAPTURE)
+	int32_t ret;
+
+	ret = iio_trig_disable(ad2s1210_hw_trig_desc);
+	if (ret) {
+		return ret;
+	}
+#endif
+	return 0;
+}
+
+int32_t ad2s1210_trigger_handler(struct iio_device_data *iio_dev_data)
```

Comment:
Comments above function definition seem to be missing here

Reply:
ok